src/cloud-providers/gcp/provider.go

```diff
@@ -0,0 +1,191 @@
+			Name: proto.String(instanceName),
+			Disks: []*computepb.AttachedDisk{
+				{
+					InitializeParams: &computepb.AttachedDiskInitializeParams{
+						DiskSizeGb:  proto.Int64(20),
+						SourceImage: proto.String(fmt.Sprintf("projects/%s/global/images/%s", p.serviceConfig.ProjectId, p.serviceConfig.ImageName)),
+						DiskType:    proto.String(fmt.Sprintf("zones/%s/diskTypes/pd-standard", p.serviceConfig.Zone)),
+					},
+					AutoDelete: proto.Bool(true),
+					Boot:       proto.Bool(true),
+					Type:       proto.String(computepb.AttachedDisk_PERSISTENT.String()),
+				},
+			},
+			Metadata: &computepb.Metadata{
+				Items: []*computepb.Items{
+					{
+						Key:   proto.String("user-data"),
+						Value: proto.String(userDataEnc),
+					},
+					{
+						Key:   proto.String("user-data-encoding"),
+						Value: proto.String("base64"),
+					},
+				},
+			},
+			MachineType: proto.String(fmt.Sprintf("zones/%s/machineTypes/%s", p.serviceConfig.Zone, p.serviceConfig.MachineType)),
+			NetworkInterfaces: []*computepb.NetworkInterface{
+				{
+					AccessConfigs: []*computepb.AccessConfig{
+						{
+							Name:        proto.String("External NAT"),
+							NetworkTier: proto.String("PREMIUM"),
+						},
+					},
+					StackType: proto.String("IPV4_Only"),
+					Name:      proto.String(p.serviceConfig.Network),
+				},
+			},
+		},
+	}
+	op, err := p.instancesClient.Insert(ctx, insertReq)
+	if err != nil {
+		return nil, fmt.Errorf("Instances.Insert error: %s. req: %v", err, insertReq)
+	}
+	err = op.Wait(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("waiting for Instances.Insert error: %s. req: %v", err, insertReq)
+	}
+	logger.Printf("created an instance %s for sandbox %s", instanceName, sandboxID)
+
+	getReq := &computepb.GetInstanceRequest{
+		Project:  p.serviceConfig.ProjectId,
+		Zone:     p.serviceConfig.Zone,
+		Instance: instanceName,
+	}
+
+	instance, err := p.instancesClient.Get(ctx, getReq)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get instance: %w, req: %v", err, getReq)
+	}
+	logger.Printf("instance name %s, id %d", instance.GetName(), instance.GetId())
+
+	ips, err := getIPs(instance)
+	if err != nil {
+		logger.Printf("failed to get IPs for the instance: %v", err)
+		return nil, err
+	}
+
+	return &provider.Instance{
+		ID:   instance.GetName(),
+		Name: instance.GetName(),
+		IPs:  ips,
+	}, nil
+}
+
+func (p *gcpProvider) DeleteInstance(ctx context.Context, instanceID string) error {
+	req := &computepb.DeleteInstanceRequest{
+		Project:  p.serviceConfig.ProjectId,
+		Zone:     p.serviceConfig.Zone,
+		Instance: instanceID,
+	}
+	op, err := p.instancesClient.Delete(ctx, req)
+	if err != nil {
+		return fmt.Errorf("Instances.Delete error: %w, req: %v", err, req)
+	}
+	err = op.Wait(ctx)
+	if err != nil {
+		return fmt.Errorf("waiting for Instances.Delete error: %s. req: %v", err, req)
+	}
+	logger.Printf("deleted an instance %s", instanceID)
+	return nil
+}
+
+func (p *gcpProvider) Teardown() error {
```

Comment:
Don't know what's the use of `Teardown()`, do you? Only Vsphere provider implements this...

Reply:
Is part of the Provider Interface.